features/cellular/framework/targets/QUECTEL/BG96/QUECTEL_BG96_CellularStack.cpp

```diff
@@ -110,20 +116,29 @@ nsapi_error_t QUECTEL_BG96_CellularStack::socket_connect(nsapi_socket_t handle,
 
 void QUECTEL_BG96_CellularStack::urc_qiurc()
 {
-    int sock_id = 0;
-
+    char urc_string[MAX_QIURC_LENGTH + 1];
     _at.lock();
-    (void) _at.skip_param();
-    sock_id = _at.read_int();
-    _at.unlock();
+    int urc_string_length = _at.read_string(urc_string, sizeof(urc_string));
+    int sock_id = _at.read_int();
+    nsapi_error_t err = _at.unlock_return_error();
+
+    if (err == NSAPI_ERROR_OK) {
```

Comment:
This a matter of personal taste as not defined by coding standard - IMHO
=> Prefer flat code i.e, structures which don't creep the code to the right side.
In the end be consistent with the style used within the module
change proposal => 
if (err != NSAPI_ERROR_OK) {
    return
}